### PR TITLE
Update contribution guide

### PR DESCRIPTION
**Pull request by gztensor**
## Description

Add PR cycle diagram and requirement to base changes on devnet-ready.

## Related Issue(s)

n/a

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Other (please describe):

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,12 +2,14 @@
...
-   branch of the subtensor GitHub repository. It is recommended to start your pull request as a
-   draft initially until you are ready to have other developers actively look at it. Any
-   changes to pallet/runtime code should be accompanied by integration and/or unit tests fully
-   testing all the edge cases of this functionality, if applicable.
+![image](docs/img/pr-lifecycle.png)
+
+1. Individuals wishing to contribute to subtensor should develop their change/feature/fix based on 
+   the `devnet-ready` branch and create a [Pull Request](https://github.com/opentensor/subtensor/compare) 
+   (PR) targeting the `devnet-ready` branch of the subtensor GitHub repository. It is recommended 
+   to start your pull request as a draft initially until you are ready to have other developers
```
Comment (orriin) [inline CONTRIBUTING.md:10]:
```suggestion
   to start your pull request as a draft until you are ready to have other developers 
```

**Review by orriin — APPROVED** [review]

**Review by JohnReedV — APPROVED** [review]

**Review by open-junius — APPROVED** [review]

**Review by sam0x17 — APPROVED** [review]
looks good but we will need to upgrade diagram with new hotfix info yeah?

**Review by cuteolaf — APPROVED** [review]
